### PR TITLE
Add encodeFile as well as decodeFile* variants

### PR DESCRIPTION
**Pull request by markus1189**
Saw #623 in the HaskellWeekly newsletter and gave it a shot.

Please review ;)

### CODE REVIEW

**Comment by bergmark** [comment]
Thanks! Looks good.

@phadej mentioned using the strict encoder as well, but do we even have such a thing??

File: Data/Aeson.hs [hunk 1]
```diff
@@ -169,6 +173,18 @@ decodeStrict :: (FromJSON a) => B.ByteString -> Maybe a
...
+-- If this fails due to incomplete or invalid input, 'Nothing' is
+-- returned.
+--
+-- The input file's content must consist solely of a JSON document,
+-- with no trailing data except for whitespace.
+--
+-- This function parses immediately, but defers conversion.  See
+-- 'json' for details.
+decodeFileStrict :: (FromJSON a) => FilePath -> IO (Maybe a)
+decodeFileStrict = fmap decodeStrict . B.readFile
```
Comment (phadej) [inline Data/Aeson.hs]: I think `either` variants will be useful as well.

**Comment by markus1189** [comment]
I added the `either` versions. :+1:

**Comment by bergmark** [comment]
Great!

I think I'll cut the 1.3.0.0 release soon, I've been holding it off too long.

**Comment by bergmark** [comment]
I just noticed, these functions need to be exported as well. Maybe travis should build with -Werror for the latest GHC to spot these things.

**Comment by markus1189** [comment]
yeah that was dumb, ;)

**Comment by phadej** [comment]
Writing tests also helps!

**Comment by markus1189** [comment]
@phadej True :smile: I can add some tests, no problem, although I was too dumb to find tests for things like `decodeStrict'` and therefore didn't spend too much effort on it.

**Comment by bergmark** [comment]
👍 
I'm fine merging this without tests, but adding them in another PR would be very welcome.

**Pull request merged**